Source/WebCore/dom/Document.cpp

```diff
@@ -9290,6 +9290,14 @@ float Document::deviceScaleFactor() const
     return deviceScaleFactor;
 }
 
+#if ENABLE(DARK_MODE_CSS)
+OptionSet<ColorScheme> Document::colorScheme(const RenderStyle* style) const
+{
+    bool isNormal = !style || style->colorScheme().isNormal();
+    return isNormal ? m_colorScheme : style->colorScheme().colorScheme();
```

Comment:
Can you update `Document::useDarkAppearance` to call this method, rather than duplicate the logic.

Reply:
Not sure I understand. There shouldn't be any ref-ing required here, nor do I see any.

Reply:
Ah, I should use `CheckedPtr` instead!

Reply:
No, that's wrong.

Reply:
Never mind. Safer C++ doesn't complaint about it.